pkg/cidr/cidr.go

```diff
@@ -135,3 +135,55 @@ func MustParseCIDR(str string) *CIDR {
 	}
 	return c
 }
+
+// Version returns the IP version for an IP, or 0 if the IP is not valid.
+func IPVersion(i *net.IP) int {
+	if i.To4() != nil {
+		return 4
+	} else if len(*i) == net.IPv6len {
+		return 6
+	}
+	return 0
+}
+
+// matchCIDRs matchs an IP address against a list of cidrs.
+// If the list is empty, it always matches.
+func matchCIDRs(ip net.IP, cidrs []*net.IPNet) bool {
+	if len(cidrs) == 0 {
+		return true
+	}
+	for _, cidr := range cidrs {
+		if cidr.Contains(ip) {
+			return true
+		}
+	}
+	return false
+}
+
+func DetectCIDRByCIDR(cidr *CIDR, version int) (*net.Interface, net.IP, *net.IPNet, error) {
+	netIfaces, err := net.Interfaces()
+	if err != nil {
+		return nil, nil, nil, fmt.Errorf("Can't detect the CIDR: Failed to enumerate interfaces")
```

Comment:
This is currently obscuring the real underlying error, please wrap errors instead:

```suggestion
		return nil, nil, nil, fmt.Errorf("Can't detect the CIDR: Failed to enumerate interfaces: %w", err)
```

This is not the only such case, please inspect all error paths to avoid losing errors.